nixos/modules/services/misc/gotenberg.nix

```diff
@@ -0,0 +1,224 @@
+          description = "Maximum queue size for LibreOffice-based conversions. Setting to 0 disables the limit.";
+        };
+
+        autoStart = mkOption {
+          type = types.bool;
+          default = false;
+          description = "Automatically start LibreOffice when Gotenberg starts. If false, Chromium will start on the first conversion request that uses it.";
+        };
+
+        disableRoutes = mkOption {
+          type = types.bool;
+          default = false;
+          description = "Disable all routes allowing LibreOffice-based conversion.";
+        };
+      };
+
+      pdfEngines = mkOption {
+        type = types.nullOr (
+          types.listOf (
+            types.enum [
+              "pdftk"
+              "qpdf"
+              "libreoffice-pdfengine"
+              "exiftool"
+              "pdfcpu"
+            ]
+          )
+        );
+        default = null;
+        description = ''
+          PDF engines to enable. Each one can be used to perform a specific task.
+          See [the documentation](https://gotenberg.dev/docs/configuration#pdf-engines) for more details.
+        '';
+      };
+
+      logLevel = mkOption {
+        type = types.enum [
+          "error"
+          "warn"
+          "info"
+          "debug"
+        ];
+        default = "info";
+        description = "The logging level for Gothenberg.";
+      };
+
+      environmentFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = "Environment file to load extra environment variables from.";
+      };
+
+      extraArgs = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = "Any extra command-line flags to pass to the Gotenberg service.";
+      };
+    };
+  };
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.enableBasicAuth -> cfg.environmentFile != null;
+        message = ''
+          When enabling HTTP Basic Authentication with `services.gotenberg.enableBasicAuth`,
+          you must provide an environment file(via `services.gotenberg.environmentFile`) with the appropriate environment variables set in it!
+
+          See `services.gotenberg.enableBasicAuth` for the names of those variables!
+        '';
+      }
+    ];
+    systemd.services.gotenberg = {
+      description = "Gotenberg API server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      path = [ cfg.package ];
+      environment = {
+        LIBREOFFICE_BIN_PATH = "${cfg.libreofficePackage}/lib/libreoffice/program/soffice.bin";
+        CHROMIUM_BIN_PATH = lib.getExe cfg.chromiumPackage;
+        FONTCONFIG_FILE = pkgs.makeFontsConf {
+          fontDirectories = [ pkgs.liberation_ttf_v2 ] ++ cfg.extraFonts;
+        };
+      };
+      serviceConfig =
+        {
+          Type = "simple";
+          DynamicUser = true;
+          ExecStart = "${lib.getExe cfg.package} ${lib.escapeShellArgs args}";
```

Comment:
some permission hardening would be nice here. these may or may not work, but as you're using DynamicUser they *probably* do

```suggestion
          ExecStart = "${lib.getExe cfg.package} ${lib.escapeShellArgs args}";
          
          # hardening options
          PrivateDevices = true;
          PrivateIPC = true;
          PrivateUsers = true;
 
          ProtectClock = true;
          ProtectControlGroups = true;
          ProtectHome = true;
          ProtectHostname = true;
          ProtectKernelLogs = true;
          ProtectKernelModules = true;
          ProtectKernelTunables = true;
          ProtectProc = "invisible";
          ProcSubset = "pid";
          
          RestrictAddressFamilies = [
            "AF_UNIX"
            "AF_INET"
            "AF_INET6"
            "AF_NETLINK"
          ];
          RestrictNamespaces = true;
          RestrictRealtime = true;
          
          LockPersonality = true;
          MemoryDenyWriteExecute = true;
          
          SystemCallFilter = [
            "@system-service"
            "~@resources"
            "~@privileged"
          ];
          SystemCallArchitectures = "native";
 
          UMask=0077
```

Reply:
Ty for the hardening options, forgot to add them.